lib/ramble/ramble/spack_runner.py

```diff
@@ -435,10 +443,16 @@ def apply_configs(self):
             'add'
         ]
 
+        active_stream = ramble.util.logger.logger.active_stream()
         for config in self.configs:
             args = config_args.copy()
             args.append(config)
-            self.spack(*args)
+
+            if active_stream is not None:
```

Comment:
Why is it checking for an active stream here, but that check isn't done on the other calls to self.spack that also pass the active stream?

Reply:
Because this code (apply_configs) is sometimes called outside of an active stream so it needs to handle the case that we have no steam nicely (and returning sys.stdout wasn't working for what this code needed).